### PR TITLE
Cleanup worker

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: src/services/data_model_engine.js [hunk 1]
```diff
@@ -274,6 +274,14 @@ export default class DataModelEngine {
     await this.bundleRepository.setBundleRepository(bundleId, BundleStatuses.sheltered, {holdUntil: new Date(bundleExpirationDate)});
   }
 
+  /**
+   * @returns {Promise<void>}: removed bundles count
```
Comment (krzysztof-jelski) [inline src/services/data_model_engine.js]: I think it's `Promise<int>` if it returns count.

File: src/services/bundle_repository.js [hunk 2]
```diff
@@ -124,11 +124,32 @@ export default class BundleRepository {
...
+    await this.db.collection('bundle_metadata').updateMany({
+      $or: [
+        {repository: {$exists: false}},
+        {'repository.holdUntil': {$lt: new Date()}}
+      ]
+    }, {$set: {repository: {status: BundleStatuses.cleanup}}});
+  }
+
+  async cleanupBundles() {
+    const cursor = await this.db.collection('bundle_metadata').find({'repository.status': BundleStatuses.cleanup}, {projection: {bundleId: 1, _id: 0}});
```
Comment (Anze1m) [inline src/services/bundle_repository.js]: Maybe also find `expendable` ones? Just in case they weren't properly deleted.

Comment (vanruch) [reply src/services/bundle_repository.js]: Amount of expendables will increase with each iteration. Also expendable status is set after bundle was removed, so realistically I don't see a case when this happens

File: src/workers/cleanup_worker.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      return;
+    }
+    try {
+      const removedBundlesCount = await this.dataModelEngine.cleanupOutdatedBundles();
+      this.logger.info(`Removed ${removedBundlesCount} outdated bundles`);
+    } finally {
+      await this.workerTaskTrackingRepository.finishWork(workId);
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (vanruch) [reply src/workers/cleanup_worker.js]: Starting up a worker is easy, but including it into atlas worker would increase complexity significantly

File: src/services/data_model_engine.js (hunk 1 above)
Comment (vanruch) [reply src/services/data_model_engine.js]: 👍

File: src/services/bundle_repository.js (hunk 2 above)
Comment (Anze1m) [reply src/services/bundle_repository.js]: Makes sense

File: src/workers/cleanup_worker.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Anze1m) [reply src/workers/cleanup_worker.js]: We can somehow, discreetly inject whole cleanup mechanism into atlas/hermes periodic work, while adding another worker potentially makes deployment and maintenance much more difficult. And also will require a lot of changes in ops repo.

[the remark this answers is not in the document]
Comment (vanruch) [reply src/workers/cleanup_worker.js]: It starts inside start_atlas, so no changes are required

**Pull request merged**